render-ws-java-client/src/main/java/org/janelia/render/client/multisem/ExponentialFitClient.java

```diff
@@ -0,0 +1,277 @@
+		public String targetStack;
+		@Parameter(names = "--coefficientsFile", description = "File name for storing the estimated coefficients in the csv format; If not given, coefficients are not stored")
+		private String coefficientsFile = null;
+		@Parameter(names = "--averageOverLayer", description = "If true, average all estimated models and apply the average to the tiles (outliers are filtered)", arity = 0)
+		public boolean averageOverLayer = true;
+		@Parameter(names = "--completeTargetStack", description = "Complete the target stack after fitting", arity = 0)
+		public boolean completeTargetStack = false;
+	}
+
+
+	private final Parameters params;
+
+	public static void main(final String[] args) {
+		final ClientRunner clientRunner = new ClientRunner(args) {
+			@Override
+			public void runClient(final String[] args) throws IOException {
+
+				final Parameters parameters = new Parameters();
+				parameters.parse(args);
+				LOG.info("runClient: entry, parameters={}", parameters);
+
+				final ExponentialFitClient client = new ExponentialFitClient(parameters);
+				client.process();
+			}
+		};
+		clientRunner.run();
+	}
+
+	public ExponentialFitClient(final Parameters parameters) {
+		this.params = parameters;
+	}
+
+	public void process() throws IOException {
+		LOG.info("process: entry");
+		final RenderDataClient renderClient = params.renderWeb.getDataClient();
+		final List<Double> zValues = renderClient.getStackZValues(params.stack);
+
+		final StackMetaData stackMetaData = renderClient.getStackMetaData(params.stack);
+		renderClient.setupDerivedStack(stackMetaData, params.targetStack);
+
+		final PrintWriter writer = getWriterIfDesired();
+		if (writer != null) {
+			writer.println("tileId,a,b,c");
+		}
+
+		for (final double z : zValues) {
+			final ResolvedTileSpecCollection tileSpecs = renderClient.getResolvedTiles(params.stack, z);
+			final Map<String, double[]> coefficients = estimateCoefficients(tileSpecs);
+
+			if (writer != null) {
+				LOG.info("process: writing coefficients for z={} to file {}", z, params.coefficientsFile);
+				appendCoefficients(coefficients, writer);
+			}
+
+			final Map<String, double[]> filterCoefficients = convertToMultiplicativeFactors(coefficients);
+			applyExponentialCorrection(tileSpecs, filterCoefficients);
+
+			renderClient.saveResolvedTiles(tileSpecs, params.targetStack, z);
+		}
+
+		if (writer != null) {
+			writer.close();
+		}
+		
+		if (params.completeTargetStack) {
+			renderClient.setStackState(params.targetStack, StackMetaData.StackState.COMPLETE);
+		}
+	}
+
+	private PrintWriter getWriterIfDesired() throws FileNotFoundException {
+		if (params.coefficientsFile == null) {
+			return null;
+		}
+		final File coefficientsFile = new File(params.coefficientsFile);
+		if (coefficientsFile.exists()) {
+			LOG.error("process: coefficients file {} already exists", params.coefficientsFile);
+			System.exit(1);
```

Comment:
I would just throw an IllegalArgumentException (or IOException if that reduces throws statements) here rather than calling System.exit.

Reply:
You're right, thanks! I still tend to think in terms of local execution.